File: LeetCode Solutions/Problems/2207. Maximize Number of Subsequences in a String.py
class Solution:
    def maximumSubsequenceCount(self, text: str, pattern: str) -> int:
        result = count1 = count2 = 0
        for c in text:
            if c == pattern[1]:
                result += count1
                count2 += 1
            if c == pattern[0]:
                count1 += 1
        return result + max(count1, count2)

File: LeetCode Solutions/Problems/2207. Maximize Number of Subsequences in a String.java
class Solution {
    public long maximumSubsequenceCount(String text, String pattern) {
        long result = 0, count1 = 0, count2 = 0;
        for (int i = 0; i < text.length(); ++i) {   
            if (text.charAt(i) == pattern.charAt(1)) {   
                result += count1;
                count2++;
            }
            if (text.charAt(i) == pattern.charAt(0)) {   
                count1++;
            }
        }
        return result + Math.max(count1, count2);
    }
}